### PR TITLE
Add email_to / email_from + simplification

### PR DESCRIPTION
**Pull request by timja**
### JIRA link (if applicable) ###
CHG0034613

### Change description ###
Email to / email from configuration was missing in production causing defaults to be taken, this sets it up so that it's correctly pulled from vault.

I've updated chart-java to allow for templating of env variables, to reduce a lot of duplication

and moved non secret values to env vars to make debugging easier

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: charts/sscs-tribunals-api/values.preview.template.yaml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 java:
   image: ${IMAGE_NAME}
   ingressHost: ${SERVICE_FQDN}
+  aadIdentityName:
   environment:
     PDF_API_URL: "http://cmc-pdf-service-aat.service.core-compute-aat.internal"
     CORE_CASE_DATA_API_URL: "http://ccd-data-store-api-aat.service.core-compute-aat.internal"
-    IDAM_OAUTH2_REDIRECT_URL: "https://evidence-sharing-preprod.sscs.reform.hmcts.net"
     DOCUMENT_MANAGEMENT_URL: "http://dm-store-aat.service.core-compute-aat.internal"
+    IDAM_S2S_AUTH: "http://rpe-service-auth-provider-aat.service.core-compute-aat.internal"
```
Comment (adusumillipraveen) [inline charts/sscs-tribunals-api/values.preview.template.yaml]: We don't need these i believe? 
```suggestion
```

File: charts/sscs-tribunals-api/values.preview.template.yaml [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
+    IDAM_URL: https://idam-api.aat.platform.hmcts.net
```
Comment (adusumillipraveen) [inline charts/sscs-tribunals-api/values.preview.template.yaml]:
```suggestion
```

Comment (timja) [reply charts/sscs-tribunals-api/values.preview.template.yaml]: this needs to be here as we don't want to talk to the preview idam

File: charts/sscs-tribunals-api/values.preview.template.yaml (hunk 1 above)
Comment (timja) [reply charts/sscs-tribunals-api/values.preview.template.yaml]: we do as env would be set to preview in the preview env?

**Comment by timja** [comment]
> This might be a breaking change till we bump chart version in flux. But that's fine as its not live

why? flux won't pick up a change automatically

File: charts/sscs-tribunals-api/values.preview.template.yaml (hunk 1 above)
Comment (adusumillipraveen) [reply charts/sscs-tribunals-api/values.preview.template.yaml]: It is set to aat in jenkins as we read aat vaults for preview.

**Pull request merged**